### PR TITLE
Added HalfGaussianVertex and tests

### PR DESCRIPTION
**Pull request by robertperrotta**
Using same concepts as HalfCauchyVertex

### CODE REVIEW

File: keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/HalfGaussianVertex.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    /**
+     * One sigma that match a proposed tensor shape of Gaussian
+     * <p>
+     * If provided parameter is scalar then the proposed shape determines the shape
+     *
+     * @param tensorShape the desired shape of the tensor in this vertex
+     * @param sigma       the sigma of the Gaussian with either the same tensorShape as specified for this vertex or a scalar
+     */
+    public HalfGaussianVertex(int[] tensorShape, DoubleVertex sigma) {
+        super(tensorShape, 0.0, sigma);
```
Comment (GeorgeNash) [inline keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/HalfGaussianVertex.java]: Can we pull all these zeroes out into a `public static final double MU = 0.0`

File: keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/HalfGaussianVertexTest.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static io.improbable.keanu.vertices.dbl.probabilistic.ProbabilisticDoubleTensorContract.moveAlongDistributionAndTestGradientOnARangeOfHyperParameterValues;
+import static org.junit.Assert.assertEquals;
+
+public class HalfGaussianVertexTest {
```
Comment (GeorgeNash) [inline keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/HalfGaussianVertexTest.java]: Can we change all the 

```
GaussianVertex tensorGaussianVertex = new HalfGaussianVertex(1);
```

to:


```
HalfGaussianVertex tensorGaussianVertex = new HalfGaussianVertex(1);
```

Comment (GeorgeNash) [inline keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/HalfGaussianVertexTest.java]: Sweet tests man - great coverage

File: keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/HalfGaussianVertex.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    public HalfGaussianVertex(double sigma) {
+        super(0.0, new ConstantDoubleVertex(sigma));
+    }
+
+    public HalfGaussianVertex(int[] tensorShape, double sigma) {
+        super(tensorShape, 0.0, new ConstantDoubleVertex(sigma));
+    }
+
+    @Override
+    public double logProb(DoubleTensor value) {
```
Comment (GeorgeNash) [inline keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/HalfGaussianVertex.java]: Nice work here. Will `HalfGaussian` only ever be used with a zeroed `Mu`, or will people want a `HalfGaussian` starting at `2.5` etc? I'll take your opinion on this as gospel

Comment (robertperrotta) [reply keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/HalfGaussianVertex.java]: Always zero for consistency with ExponentialVertex (we can always add a constant to it with the same effect as a nonzero location).

File: keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/HalfGaussianVertexTest.java (hunk 2 above)
Comment (robertperrotta) [reply keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/HalfGaussianVertexTest.java]: Yes.

**Pull request merged**